### PR TITLE
[New] `display`: add `contents`

### LINKED ISSUE #32: Support display="contents"
By using `display="contents"` we can avoid having the div, what is generated by `react-outside-click-handler`, having influence on the visual appearance of it's children. This would be of immense value since this lib is only a helper and should not mess with the layout as such.

Proposal: Allow `display` to be set to `contents` OR remove the constraint on the `display`-property completely (it could be of the users choice what value to pass, just mention in the docs what is supported officially).

Some more details about `display="contents"`: http://adrianroselli.com/2018/05/display-contents-is-not-a-css-reset.html

### PR DESCRIPTION
**Pull request by Raveline**
Fixes #32.

I noticed in #28 you requested adding tests. I'm willing to add some, but I'd like to know what exactly one should test for: proper display of children, or proper behaviour of the outside click (I imagine by mimicking the "when `this.childNode` does not contain `e.target`" case, but simply with a OutsideClickHandler with the display set to `CONTENTS` ?).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,6 +42,6 @@ See https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Ev
 
 If `useCapture` is true, the event will be registered in the capturing phase and thus, propagated top-down instead of bottom-up as is the default.
 
-### display: `PropTypes.oneOf(['block', 'flex', 'inline-block'])`
+### display: `PropTypes.oneOf(['block', 'flex', 'inline-block', 'contents'])`
 
-By default, the `OutsideClickHandler` renders a `display: block` `<div />` to wrap the subtree defined by `children`. If desired, the `display` can be set to `inline-block` or `flex` instead. There is no way not to render a wrapping `<div />`.
+By default, the `OutsideClickHandler` renders a `display: block` `<div />` to wrap the subtree defined by `children`. If desired, the `display` can be set to `inline-block`, `flex` or `contents` instead. There is no way not to render a wrapping `<div />`.
```
Comment (ljharb) [inline README.md]:
```suggestion
By default, the `OutsideClickHandler` renders a `display: block` `<div />` to wrap the subtree defined by `children`. If desired, the `display` can be set to `inline-block`, `flex`, or `contents` instead. There is no way not to render a wrapping `<div />`.
```

**Comment by ljharb** [comment]
If you want to add "inline" here as well, along with tests, and subsume #28, that's also fine.

**Comment by Raveline** [comment]
I added some tests. I'm not 100% sure these are what you expected because they do seem a bit redundant, but they will at least offer a guarantee of no-regression on supported styles.

I also took the opportunity to subsume #28 since it was trivial. Tell me if there is anything more this needs.

File: test/OutsideClickHandler_test.jsx [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+import { use, expect } from 'chai';
 import sinon from 'sinon-sandbox';
 import { shallow, mount } from 'enzyme';
 import wrap from 'mocha-wrap';
 import contains from 'document.contains';
-
+import chaiEnzyme from 'chai-enzyme';
 import OutsideClickHandler from '../src/OutsideClickHandler';
 
+use(chaiEnzyme());
```
Comment (ljharb) [inline test/OutsideClickHandler_test.jsx]: not so sure this complexity pays for itself in these two tests

Comment (Raveline) [reply test/OutsideClickHandler_test.jsx]: Yes, I hesitated, but it was either have a complex way of writing the test at the cost of readability, or adding complexity through import. I opted for the second option, in the hope that future tests could benefit from chai-enzyme.

Comment (ljharb) [reply test/OutsideClickHandler_test.jsx]: I ended up removing this; it's a one-liner with basic chai, so there's zero readability hit.

**Comment by delijah** [comment]
Will you release a new version soon (which includes that fix)?

**Pull request merged**